### DISCUSSION TITLE
Using an alternative http client?

### DISCUSSION THREAD

**Discussion by jolivaSan** [panva/oauth4webapi#20] [Q&A]
I have read the docs and seems like to execute the http request the library use fetch, is any way to provide an alternative client? We were looking for a client that support timeouts, retries... as for example https://github.com/sindresorhus/ky.

Best regards

**Comment by panva** [comment]
Retry functionality is something you can wrap around a given routine and timeouts you can manage through AbortSignal instances passed to the individual routines that execute http requests. 

I do not plan to support http client swapping.

**Reply by jolivaSan** [reply]
I understand that if i want to retry for example the method oauth.authorizationCodeGrantRequest() based on status codes i can do it, but how can i abort the fetch that is perform inside the library with the AbortSignal? as long as I know to abort a fetch you need to pass the signal of the abort controller, if not the fetch request will continue... is there any way to pass configuration to the fetchs operations used by the library?

**Reply by panva** [reply]
> how can i abort the fetch that is perform inside the library with the AbortSignal

every routine which may end up making an http request accepts an options argument where an optional `signal` property can be passed.

**Comment by panva** [comment]
I'm looking for feedback for this request in #94
